src/Device/Driver/XCVario.cpp

```diff
@@ -196,10 +196,12 @@ bool
 XVCDevice::PutBallast(double fraction, gcc_unused double overload,
                       OperationEnvironment &env)
 {
-  /* the XCVario understands the CAI302 "!g" command for
-     MacCready, ballast and bugs */
-
-  return CAI302::PutBallast(port, fraction, env);
+  /* the XCVario understands CAI302 like command for ballast "!g,b" with
+     float precision */
+   char buffer[32];
+   float ballast = fraction * 10.;
```

Comment:
Wrong type. This narrows the `double` to `float`, and the `sprintf()` call will widen it again to `double`. This not only loses precision (which doesn't matter here), but also adds overhead.